src/components/Card.js

```diff
@@ -1,41 +1,99 @@
+    handleCardClick,
+    handleCardDeleteFunctionInIndexComponent
+  ) {
     this._cardData = cardData;
+    this._id = cardData._id;
     this._name = cardData.name;
     this._link = cardData.link;
+    this._owner = cardData.owner._id;
+    this._cardId = cardData._id;
+    this._likes = cardData.likes;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleCardDeleteFunctionInIndexComponent =
+      handleCardDeleteFunctionInIndexComponent;
   }
 
-  _handleFavIcon = () => {
-    this._favIconElement.classList.toggle("card__fav-icon-selected");
+  _handleFavIconClick = () => {
+    //variable to register if current user liked the image or not
+    const currentUserLike = this._likes.find((user) => {
+      return user._id === "f50447686616d1fa985ca0e1";
+    });
+    let didCurrentUserLikeThisCard = currentUserLike ? true : false;
+    if (didCurrentUserLikeThisCard) {
+      // 1. Make a call to network to remove the users like for this card
+      api.unLikeACard(this._cardId).then((data) => {
+        this._likes = data.likes;
+        this._favIconElement.classList.remove("card__fav-icon-selected");
+        this._favCountElement.textContent = this._likes.length;
+      });
+    } else {
+      // 1. Make a call to network to add the users like for this card
+      api.likeACard(this._cardId).then((data) => {
+        this._likes = data.likes;
+        // 2. Update state of the like button and update counter
+        this._favIconElement.classList.toggle("card__fav-icon-selected");
+        this._favCountElement.textContent = this._likes.length;
+      });
+    }
+  };
+
+  _handleDeleteCardFunctionInCardComponent = () => {
+    this.deleteCardPopup = new DeleteCardForm(
+      "#delete-image-confirm-modal",
+      this._handleDeleteCardFormSubmit
+    );
+    this.deleteCardPopup.openModal();
   };
 
-  _handleDelButton = () => {
+  _handleDeleteCardFormSubmit = () => {
     this._cardElement.remove();
-    this._cardElement = null;
+    this._handleCardDeleteFunctionInIndexComponent(this._cardId);
+    if (this.deleteCardPopup) {
+      this.deleteCardPopup.closeModal();
+    }
   };
 
   _onCardClick = (ev) => {
     this._handleCardClick(this._cardData);
   };
 
-  _setEventListeners() {
-    this._favIconElement.addEventListener("click", this._handleFavIcon);
-    this._deleteCardButton.addEventListener("click", this._handleDelButton);
-    this._cardElement
-      .querySelector(".card__image")
-      .addEventListener("click", this._onCardClick);
-  }
-
   getCardElement() {
     this._cardTemplate = document.querySelector(this._cardSelector);
     this._cardElement = this._cardTemplate.content
       .querySelector(".card")
       .cloneNode(true);
-    this._deleteCardButton =
-      this._cardElement.querySelector(".card__del-button");
+    this._deleteCardIcon = this._cardElement.querySelector(".card__del-button");
+    if (this._owner === "f50447686616d1fa985ca0e1") {
```

Comment:
* `_id` of the user should be received from the server only 1 time. You need to store it in a variable `userId` or in class `UserInfo` and use it when you create a card passing it to the constructor of `Card`